lib/HLSL/DxilValidation.cpp

```diff
@@ -1390,8 +1389,11 @@ static void ValidateGather(CallInst *CI, Value *srvHandle, Value *samplerHandle,
   default:
     // Invalid resource type for gather.
     ValCtx.EmitInstrError(CI, ValidationRule::InstrResourceKindForGather);
-    break;
+    return;
```

Comment:
Return on error to prevent calling ValidateResourceOffset on an invalid resource as happens with some of our tests specifically targeting the validator by altering DXIL after it is produced